app/models/bill.rb

```diff
@@ -0,0 +1,41 @@
+class Bill < ActiveRecord::Base
+  has_many :sponsorships
+  has_many :cosponsors, -> { merge(Sponsorship.cosponsored) }, through: :sponsorships, source: :legislator
+  has_one :authorship, -> { merge(Sponsorship.sponsored) }, class_name: 'Sponsorship'
+  has_one :sponsor, through: :authorship, source: :legislator
+
+  validates :bill_id, uniqueness: true
+
+  def self.fetch(bill_id:)
+    if results = Integration::Sunlight.get_bill(bill_id:  bill_id)
+      create_or_update(results.deep_symbolize_keys)
+    end
+  end
+
+  def self.create_or_update(hash)
+    bill_id       = hash.delete(:bill_id)
+    sponsor_id    = hash.delete(:sponsor_id)
+    introduced_at = hash.delete(:introduced_at)
+    cosponsors    = hash.delete(:cosponsors) || []
+    bill = find_or_initialize_by(bill_id: bill_id).tap{|b| b.update(hash)}
```

Comment:
Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.